docs/changelog.rst

```diff
@@ -11,6 +11,12 @@ KSQL 5.2 includes new features, including:
 * Added support for read-after-write consistency: new commands don't execute until previous commands have finished executing.
   This feature is enabled by default in the CLI (`#2280 <https://github.com/confluentinc/ksql/pull/2280>`_)
   and can be implemented by the user for the REST API (:ref:`coordinate_multiple_requests`).
+* Automatic compatibility management for queries in headless mode across versions. Starting with 5.2, KSQL will automatically take care
```

Comment:
I think it might also make sense to call out the config topic in the main docs, but I was a bit unsure about which section that would fit into. Any ideas @JimGalasyn ?

Reply:
Seems like the [KSQL Query Settings](https://docs.confluent.io/current/ksql/docs/installation/server-config/config-reference.html#ksql-query-settings) section would work. You could create an anchor there and link to it, something like:
```
.. _ksql-server-config-query-settings:
```

Reply:
Not sure the query settings page is the right place since the config topic Rohan wants to add a docs section about isn't query specific, rather it will be created automatically (starting in version 5.2) for each headless mode deployment. Similar to how interactive deployments automatically create an internal command topic (the config topic is also internal). If there are docs describing the internals of KSQL (e.g., anywhere the command topic is mentioned), that could be a good place. I can't seem to find one though.

I'd suggest adding it into [Non-interactive (Headless) KSQL Usage](https://docs.confluent.io/current/ksql/docs/installation/server-config/index.html#non-interactive-headless-ksql-usage) except that article is part of the "Configuring KSQL Server" page and doesn't feel like the right place to discuss KSQL internals.

cc @JimGalasyn

Reply:
Hmm, maybe the [KSQL Architecture](https://docs.confluent.io/current/ksql/docs/concepts/ksql-architecture.html) will work.